Build failed if UA_ENABLE_TYPEDESCRIPTION disabled

<!--
!ATTENTION!
Please read the following page carefully and provide us with all the
information requested:
https://github.com/open62541/open62541/wiki/Writing-Good-Issue-Reports

Use Github Markdown to format your text:
https://help.github.com/articles/basic-writing-and-formatting-syntax/

Fill out the sections and checklist below (add text at the end of each line).

!ATTENTION!
--------------------------------------------------------------------------------
-->

## Description

When try disable `UA_ENABLE_TYPEDESCRIPTION` build failed.

## Background Information / Reproduction Steps

Used CMake options:

<!-- 

Include all CMake options here, which you modified or used for your build.

If you are using cmake-gui, go to "Tools > Show my Changes" and paste the content of "Command Line Options"

On the command line use `cmake -L` (or `cmake -LA` if you changed advanced variables)
-->

```bash
cmake -DUA_ENABLE_TYPEDESCRIPTION=0 ..
```

## Checklist

I can reproduce error in master branch and 1.5 Release and 1.4 Release.

Error log example in master:
```
/home/ud/src/open62541/src/ua_types_definition.c: In function ‘UA_DataType_fromEnumDescription’:
/home/ud/src/open62541/src/ua_types_definition.c:393:14: error: ‘UA_DataTypeMember’ has no member named ‘memberName’; did you mean ‘memberType’?
  393 |         dtm->memberName = (char*)UA_malloc(ef->name.length + 1);
      |              ^~~~~~~~~~
      |              memberType
In file included from /home/ud/src/open62541/include/open62541/types.h:21,
                 from /home/ud/src/open62541/src/ua_types_definition.c:9:
/home/ud/src/open62541/src/ua_types_definition.c:394:23: error: ‘UA_DataTypeMember’ has no member named ‘memberName’; did you mean ‘memberType’?
  394 |         UA_CHECK(dtm->memberName != NULL,
      |                       ^~~~~~~~~~
/home/ud/src/open62541/build/src_generated/open62541/config.h:558:43: note: in definition of macro ‘UA_UNLIKELY’
  558 | # define UA_UNLIKELY(x) __builtin_expect((x), 0)
      |                                           ^
/home/ud/src/open62541/src/ua_types_definition.c:394:9: note: in expansion of macro ‘UA_CHECK’
  394 |         UA_CHECK(dtm->memberName != NULL,
      |         ^~~~~~~~
/home/ud/src/open62541/src/ua_types_definition.c:396:39: error: ‘UA_DataTypeMember’ has no member named ‘memberName’; did you mean ‘memberType’?
  396 |         memcpy((char*)(uintptr_t)dtm->memberName, ef->name.data, ef->name.length);
      |                                       ^~~~~~~~~~
      |                                       memberType
/home/ud/src/open62541/src/ua_types_definition.c:397:34: error: ‘UA_DataTypeMember’ has no member named ‘memberName’; did you mean ‘memberType’?
  397 |         *(char*)(uintptr_t)&dtm->memberName[ef->name.length] = '\0';
      |                                  ^~~~~~~~~~
      |                                  memberType
[ 17%] Building C object CMakeFiles/open62541-plugins.dir/arch/posix/eventloop_posix_tcp.c.o
```

--- a/src/ua_types_definition.c
+++ b/src/ua_types_definition.c
@@ -391,11 +391,13 @@ UA_DataType_fromEnumDescription(UA_DataType *type,
         UA_DataTypeMember *dtm = &type->members[i];
         const UA_EnumField *ef = &descr->enumDefinition.fields[i];
         dtm->memberType = (const UA_DataType*)(uintptr_t)ef->value;
+#ifdef UA_ENABLE_TYPEDESCRIPTION
         dtm->memberName = (char*)UA_malloc(ef->name.length + 1);
         UA_CHECK(dtm->memberName != NULL,
                  UA_DataType_clear(type); return UA_STATUSCODE_BADOUTOFMEMORY);
         memcpy((char*)(uintptr_t)dtm->memberName, ef->name.data, ef->name.length);
         *(char*)(uintptr_t)&dtm->memberName[ef->name.length] = '\0';
+#endif
     }
 
     return UA_STATUSCODE_GOOD;

